controllers/release/adapter.go

```diff
@@ -570,3 +607,31 @@ func (a *adapter) validateProcessingResources() *controller.ValidationResult {
 	}
 	return &controller.ValidationResult{Valid: true}
 }
+
+// validatePipelineRef checks that the release PipelineRun ref passes the checks from the ReleaseServiceConfig.
+func (a *adapter) validatePipelineRef() *controller.ValidationResult {
+	releasePlanAdmission, err := a.loader.GetActiveReleasePlanAdmissionFromRelease(a.ctx, a.client, a.release)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			a.release.MarkValidationFailed(err.Error())
+			return &controller.ValidationResult{Valid: false}
+		}
+		return &controller.ValidationResult{Err: err}
+	}
+
+	releaseStrategy, err := a.loader.GetReleaseStrategy(a.ctx, a.client, releasePlanAdmission)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			a.release.MarkValidationFailed(err.Error())
+			return &controller.ValidationResult{Valid: false}
+		}
+		return &controller.ValidationResult{Err: err}
+	}
+
+	if !a.releaseServiceConfig.AreDebugOptionsValid(releaseStrategy.Spec.Bundle) {
```

Comment:
This should just be `if !a.releaseServiceConfig.Debug && releaseStrategy.Spec.Bundle == ""`. No need for that function. In addition, Martin is modifying this part so we should be able to do something like:
```
<releaseStrategy|rpa|rp>.Spec.PipelineRef.UsesBundleResolver()
<releaseStrategy|rpa|rp>.Spec.PipelineRef.UsesClusterResolver()
<releaseStrategy|rpa|rp>.Spec.PipelineRef.UsesGitResolver()
```
That or something similar, as we will control the PipelineRef (it won't be the Tekton one but one defined in Operator Toolkit).

Reply:
I don't see why we don't want a helper function for this. We will have bundle == "", is pipeline in the allow list, is pipeline in the disallow list, does it use bundle resolver, does it use cluster resolver, does it use git resolver, and who knows what other checks will come up. I think that warrants a helper function

Reply:
It can be just `IsClusterScoped()`. And I'm not opposing to a helper function. I'm just saying that the config resource should not have that logic.